ui/lib/kv/addon/components/kv-delete-modal.hbs

```diff
@@ -1,16 +1,23 @@
-<button type="button" class="toolbar-link" {{on "click" (fn (mut this.modalOpen) true)}} data-test-kv-delete={{@mode}}>
-  {{yield}}
-</button>
+<Hds::Button
+  @text={{or @text (capitalize @mode)}}
+  @color="secondary"
+  class="toolbar-button"
+  {{on "click" (fn (mut this.modalOpen) true)}}
+  data-test-kv-delete={{@mode}}
+/>
+
 {{#if this.modalOpen}}
-  <Modal
-    @title={{this.modalDisplay.title}}
+  <Hds::Modal
+    id="kv-delete-modal-{{@mode}}"
+    @color={{this.modalDisplay.color}}
     @onClose={{fn (mut this.modalOpen) false}}
-    @isActive={{this.modalOpen}}
-    @type={{this.modalDisplay.type}}
-    @showCloseButton={{true}}
     data-test-delete-modal
+    as |M|
   >
-    <section class="modal-card-body">
+    <M.Header @icon="trash">
```

Comment:
Is every header going to be a trash icon or should we add an icon key to the `modalDisplay` object in the component?

Reply:
These are all delete related, so they'll all have the trash!